rosbag2_test_common/include/rosbag2_test_common/temporary_directory_fixture.hpp

```diff
@@ -17,26 +17,61 @@
 
 #include <gmock/gmock.h>
 
+#include <filesystem>
 #include <string>
 
-#include "rcpputils/filesystem_helper.hpp"
+#ifdef _WIN32
+#  include <io.h>
+#endif
 
 using namespace ::testing;  // NOLINT
 
 namespace rosbag2_test_common
 {
 
+/**
+ * @brief std::filesystem version of create_temp_directory from rcpputils/src/filesystem_helper.cpp
+ */
+std::filesystem::path create_temp_directory(
+  const std::string & base_name,
+  const std::filesystem::path & parent_path = std::filesystem::temp_directory_path())
+{
+  const auto template_path = base_name + "XXXXXX";
+  std::string full_template_str = (parent_path / template_path).generic_string();
+  std::filesystem::create_directories(parent_path);
+
+#ifdef _WIN32
+  errno_t errcode = _mktemp_s(&full_template_str[0], full_template_str.size() + 1);
+  if (errcode) {
+    std::error_code ec(static_cast<int>(errcode), std::system_category());
+    throw std::system_error(ec, "could not format the temp directory name template");
+  }
+  const std::filesystem::path final_path{full_template_str};
+  std::filesystem::create_directory(final_path);
+#else
+  const char * dir_name = mkdtemp(&full_template_str[0]);
+  if (dir_name == nullptr) {
+    std::error_code ec{errno, std::system_category()};
+    errno = 0;
+    throw std::system_error(ec, "could not format or create the temp directory");
+  }
+  const std::filesystem::path final_path{dir_name};
+#endif
+
+  return final_path;
+}
```

Comment:
I'm very sorry about this, but I now think that we should remove this and keep it in `rcpputils`.  The main reason is that I found another use for this function in https://github.com/ros2/rcl_logging/pull/115 , so even when we remove most of https://github.com/ros2/rcpputils/blob/292be119c13edb8baa1e78c348637c460ad7a846/src/filesystem_helper.cpp , we will still want to keep this function as a utility not offered by the STL.

Again, apologies for my short-sightedness.

Reply:
reverted